share/benchmark/xtp_benchmark.in

```diff
@@ -86,105 +104,131 @@ progargs = argparse.ArgumentParser(prog='xtp_benchmark',
+    os.mkdir(OPTIONS.output_folder)
+
+    input_molecules = []
+
+    pattern = "*.xyz"
+
+    for path, subdirs, files in os.walk(OPTIONS.input_folder):
+        for name in files:
+            if fnmatch.fnmatch(name, pattern):
+                input_molecules.append([path, name])
+
+    for mol in input_molecules:
+        newpath = os.path.join(OPTIONS.output_folder,
+                               os.path.splitext(mol[1])[0])
+        os.mkdir(newpath)
+        shutil.copyfile(os.path.join(
+            mol[0], mol[1]), os.path.join(newpath, "system.xyz"))
+
+    print("Created folders for:\n{}".format(
+        "\n".join(m[1] for m in input_molecules)))
 
 if OPTIONS.run:
-    print("Creating run directory")
-    if os.path.isdir('RUN'):
-        xxquit("Folder 'RUN' already exists")
-    os.mkdir("RUN")
-    CheckforXMLFile('dftgwbse.xml')
-    CheckforXMLFile('gwbse.xml')
-    CheckforXMLFile('xtpdft.xml')
-
-    molecules = []
-
-    for folder in os.listdir("Geometries_ThielSet"):
-        print("Reading geometries from folder '{}'".format(folder))
-        for geo in os.listdir("Geometries_ThielSet/"+folder):
-            if os.path.splitext("Geometries_ThielSet/"+folder+"/"+geo)[1] == ".xyz":
-                molname = os.path.splitext(geo)[0]
-                print("Creating subfolder '{}'".format(molname))
-                dest = "RUN/{}".format(molname)
-                molecules.append(molname)
-                os.mkdir(dest)
-                shutil.copyfile("Geometries_ThielSet/"+folder +
-                                "/"+geo, dest+"/system.xyz")
-                shutil.copyfile('dftgwbse.xml', dest+'/dftgwbse.xml')
-                shutil.copyfile('gwbse.xml', dest+'/gwbse.xml')
-                shutil.copyfile('xtpdft.xml', dest+'/xtpdft.xml')
 
     print("\nStarting benchmark with {} threads".format(OPTIONS.threads))
+    if HasGPUs:
+        print(" and {} GPUs".format(OPTIONS.gpus))
+
+    cmd_options = "job_name=system "
+    if OPTIONS.cmdoptions:
+        if isinstance(OPTIONS.cmdoptions, str):
+            cmd_options += OPTIONS.cmdoptions
+        else:
+            cmd_options += " ".join(OPTIONS.cmdoptions)
+    print(f"With options: '{cmd_options}'")
     print(30*"-")
 
     moldata = {}
     totaltime = time.time()
     referencedata = {}
-    if os.path.isfile("references.json"):
-        print("Loading reference data from 'references.json'")
-        with open("references.json", 'r') as fp:
+    if OPTIONS.reference:
+        print("Loading reference data from '{}'".format(OPTIONS.reference))
+        with open(OPTIONS.reference, 'r') as fp:
             referencedata = json.load(fp)
 
-    for mol in molecules:
+    for mol in os.listdir(OPTIONS.output_folder):
         print("Running {}".format(mol).ljust(30), end="\r")
-        with cd("RUN/{}".format(mol)):
-            
-            start = time.time()
+        with cd(os.path.join(OPTIONS.output_folder, mol)):
```

Comment:
I have never seen a context manager change into a directory, it is a very interesting application of context managers

Reply:
Stack overflow. I like it, it is quite convienient. :)